packages/fluentui/react-northstar/src/components/Accordion/Accordion.tsx

```diff
@@ -84,114 +95,108 @@ export interface AccordionProps extends UIComponentProps, ChildrenComponentProps
+    renderPanelContent,
+    renderPanelTitle,
+  } = props;
+  const alwaysActiveIndex = expanded ? 0 : -1;
+
+  const [activeIndex, setActiveIndex] = useAutoControlled({
+    defaultValue: props.defaultActiveIndex,
+    value: props.activeIndex,
+    initialValue: exclusive ? alwaysActiveIndex : [alwaysActiveIndex],
+  });
 
-  actionHandlers = {
+  const actionHandlers = {
     moveNext: e => {
       e.preventDefault();
-      this.focusHandler.moveNext();
+      focusHandler.moveNext();
     },
     movePrevious: e => {
       e.preventDefault();
-      this.focusHandler.movePrevious();
+      focusHandler.movePrevious();
     },
     moveFirst: e => {
       e.preventDefault();
-      this.focusHandler.moveFirst();
+      focusHandler.moveFirst();
     },
     moveLast: e => {
       e.preventDefault();
-      this.focusHandler.moveLast();
+      focusHandler.moveLast();
     },
   };
 
-  constructor(props, context) {
-    super(props, context);
+  const getA11yProps = useAccessibility<AccordionBehaviorProps>(accessibility, {
+    debugName: Accordion.displayName,
+    actionHandlers,
+    rtl: context.rtl,
+  });
+
+  const { classes } = useStyles<AccordionStylesProps>(Accordion.displayName, {
+    className: accordionClassName,
+    mapPropsToInlineStyles: () => ({
+      className,
+      design,
+      styles,
+      variables,
+    }),
+    rtl: context.rtl,
+  });
 
-    this.focusHandler = new ContainerFocusHandler(this.getNavigationItemsSize, this.handleNavigationFocus, true);
-  }
+  const [focusedIndex, setfocusedIndex] = React.useState(0);
 
-  handleNavigationFocus = (index: number) => {
-    this.setState({ focusedIndex: index }, () => {
-      const targetComponent = this.itemRefs[index] && this.itemRefs[index].current;
-      targetComponent && targetComponent.focus();
-    });
+  const handleNavigationFocus = (index: number) => {
+    setfocusedIndex(index);
   };
 
-  getNavigationItemsSize = () => this.props.panels.length;
+  React.useEffect(() => {
+    const targetComponent = itemRefs[focusedIndex] && itemRefs[focusedIndex].current;
+    targetComponent && targetComponent.focus();
+  }, [focusedIndex]);
 
-  getInitialAutoControlledState({ expanded, exclusive }: AccordionProps) {
-    const alwaysActiveIndex = expanded ? 0 : -1;
-    return { activeIndex: exclusive ? alwaysActiveIndex : [alwaysActiveIndex] };
-  }
+  const getNavigationItemsSize = () => props.panels.length;
+  const unhandledProps = useUnhandledProps(Accordion.handledProps, props);
+  const ElementType = getElementType(props);
 
-  computeNewIndex = (index: number): number | number[] => {
-    const { activeIndex } = this.state;
-    const { exclusive } = this.props;
+  const focusHandler: ContainerFocusHandler = new ContainerFocusHandler(
+    getNavigationItemsSize,
+    handleNavigationFocus,
+    true,
+  );
 
-    if (!this.isIndexActionable(index)) {
+  let itemRefs = [];
+  const defaultAccordionTitleId = _.uniqueId('accordion-title-');
```

Comment:
can you check if these get re-generated at every render? if so, maybe use a ref or memo.

Reply:
Changed to `useMemo`